packages/react-router/src/link.tsx

```diff
@@ -432,7 +433,15 @@ export type LinkOptions<
 }
 
 export type CheckPath<TRouteTree extends AnyRoute, TPass, TFail, TFrom, TTo> =
-  ResolveRoute<TRouteTree, TFrom, TTo> extends never ? TFail : TPass
+  ResolveRoute<TRouteTree, TFrom, TTo> extends infer TRoute extends AnyRoute
+    ? [TRoute] extends [never]
+      ? TFail
```

Comment:
is this really necessary?
I remove it, and type tests still work (and it reduces instantiations from 573059 to 572245)

Reply:
You're right atm and thats because we're not strictly checking `from`. But this first checks if the route exists at all and if it does not then fail.